packages/patternfly-4/react-core/src/components/Card/CardHead.tsx

```diff
@@ -15,6 +25,11 @@ export const CardHead: React.FunctionComponent<CardHeadProps> = ({
   ...props
 }: CardHeadProps) => (
   <div className={css(styles.cardHead, className)} {...props}>
-    {children}
+    {React.Children.map(children, child => {
+      if (isReactElement(child) && (child.type === CardActions || child.type === CardHeader)) {
```

Comment:
This still does not look right.  `CardHeader` should be wrapped by `CardHeadMain`.  You are also still forcing the actions to come before 'CardHeadMain` and I don't think we want to add that restriction. 
Since 'CardHeadMain` is not required,  I still think your original implementation with leaving it up to the consumer to add the wrappper is the best solution here.  We just need to make sure it is documented appropriately.